src/web/app/components/session-edit-form/session-edit-form.component.spec.ts

```diff
@@ -53,4 +67,312 @@ describe('SessionEditFormComponent', () => {
+      + 'when response visible setting is CUSTOM and submissionStartDate is before customResponseVisibleDate', () => {
+    component.model.responseVisibleSetting = ResponseVisibleSetting.CUSTOM;
+    component.model.submissionStartTime =
+        service.getTimeInstance(moment());
+    component.model.customResponseVisibleTime =
+        service.getTimeInstance(moment());
+    component.model.submissionStartDate =
+        service.getDateInstance(moment().tz(component.model.timeZone));
+    component.model.customResponseVisibleDate =
+        service.getDateInstance(moment().tz(component.model.timeZone).add(1, 'days'));
+    const maxTimeForSessionVisible = component.maxTimeForSessionVisible;
+    expect(maxTimeForSessionVisible).toEqual(component.model.submissionStartTime);
+  });
+
+  it('should return customResponseVisibleTime as the maximum time for session visibility '
+      + 'when response visible setting is CUSTOM and submissionStartDate is after customResponseVisibleDate', () => {
+    component.model.responseVisibleSetting = ResponseVisibleSetting.CUSTOM;
+    component.model.submissionStartTime =
+        service.getTimeInstance(moment());
+    component.model.customResponseVisibleTime =
+        service.getTimeInstance(moment());
+    component.model.submissionStartDate =
+        service.getDateInstance(moment().tz(component.model.timeZone).add(1, 'days'));
+    component.model.customResponseVisibleDate =
+        service.getDateInstance(moment().tz(component.model.timeZone));
+    const maxTimeForSessionVisible = component.maxTimeForSessionVisible;
+    expect(maxTimeForSessionVisible).toEqual(component.model.customResponseVisibleTime);
+  });
+
+  it('should return the default time format if response visible setting is not recognized', () => {
+    const maxTimeForSessionVisible = component.maxTimeForSessionVisible;
+    expect(maxTimeForSessionVisible).toEqual(getDefaultTimeFormat());
+  });
+
+  it('should return submissionStartDate as the minimum date for response visibility'
+    + ' when session visible setting is AT_OPEN', () => {
+    component.model.sessionVisibleSetting = SessionVisibleSetting.AT_OPEN;
+    component.model.submissionStartDate =
+        service.getDateInstance(moment().tz(component.model.timeZone));
+    const minDateForResponseVisible = component.minDateForResponseVisible;
+    expect(minDateForResponseVisible).toEqual(component.model.submissionStartDate);
+  });
+
+  it('should return customSessionVisibleDate as the minimum date for response visibility '
+      + 'when session visible setting is CUSTOM', () => {
+    component.model.sessionVisibleSetting = SessionVisibleSetting.CUSTOM;
+    component.model.submissionStartDate =
+        service.getDateInstance(moment().tz(component.model.timeZone));
+    component.model.customSessionVisibleDate =
+        service.getDateInstance(moment().tz(component.model.timeZone));
```

Comment:
we should make customerSessionVisibleDate different from submissionStartDate here